core/modules/comment/comment.install

```diff
@@ -78,6 +78,9 @@ function comment_uninstall() {
  * Implements hook_enable().
  */
 function comment_enable() {
+  // Remove orphaned comments of users that do not exist anymore.
+  $user_uids = db_select('users', 'u')->fields('u', array('uid'));
+  db_delete('comment')->condition('uid', $user_uids, 'NOT IN')->execute();
```

Comment:
We should be checking if the user was removed and all content was supposed to be assigned to Anonymous with the option: " Delete the account and make its content belong to the Anonymous user." . I'm assuming that doesn't happen for comments if it was disabled while user is deleted.

Reply:
If it's not easy to determine if the comments should be deleted or not, I think we should err on side of assigning them to Anonymous. At least in the narrow case where the comment module was disabled.

Reply:
There's no easy way to figure out, what happened while the comment module was disabled.

I'm leaning towards "delete" here as to me it seems more problematic to show comments of cancelled users as "anonymous" when all their content was actually supposed to get deleted.

Reply:
I think it's worse if we delete comments where someone intended to preserve them but just assign them to Anonymous. There's no going back.

Perhaps also check if the node is deleted and only delete them in that case. The default could be just unpublishing them.

Reply:
Regarding the comment reassign: Maybe I (or all europeans) got nuts with all that GPDR stuff. :wink: I'll think this over a bit.

Regarding the check for deleted nodes: That might make sense.

Reply:
Can we replace all comments of this sort with "this comment is not available" and/or unpublish them? Then the site admin can reinstate them if he wants.

Reply:
@docwilmot As the uid is gone, we can't handle these comments as entities anymore. (Without existing uid the comment is broken.) The PR already catches Exceptions when viewing a node. This sql in hook_enable() is just some sort of housekeeping.

Reply:
I meant as a compromise to your discussion above re delete vs reassign. Dont delete (because unintentional data loss might happen), but also dont just reassign (the deleted user may have been a spammer or an undesirable). Reassign and hide, let the user decide if to keep.

Reply:
I'd suggest to discuss this further at the issue directly. It might get too hard for other people to keep up if we discuss at two locations simultaneously.